File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.swift
import Foundation

func solution(_ s:String) -> [Int] {
    var s = s
    var previousCount = s.count
    var removedZero = 0
    var count = 0

    while(s != "1"){
        s=s.replacingOccurrences(of:"0",with:"")
        removedZero += previousCount-s.count
        s = String(s.count,radix:2)
        previousCount = s.count
        count += 1
    }
    
    return [count,removedZero]
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    count = 0
    sum = 0
    while s != '1' :
        count += 1
        removedZero = s.replace('0','')
        sum += len(s) - len(removedZero)

        s = format(len(removedZero),'b')
    
    return [count,sum]